traverson4j-hc5/src/main/java/uk/co/autotrader/traverson/http/ApacheHttpTraversonClientAdapter.java

```diff
@@ -37,32 +37,42 @@ public ApacheHttpTraversonClientAdapter(CloseableHttpClient client) {
 
     @Override
     public <T> Response<T> execute(Request request, Class<T> returnType) {
-        HttpUriRequest httpUriRequest = apacheHttpUriConverter.toRequest(request);
+        ClassicHttpRequest httpRequest = apacheHttpUriConverter.toRequest(request);
 
         HttpClientContext clientContext = HttpClientContext.create();
-        CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
+        BasicCredentialsProvider credentialsProvider = new BasicCredentialsProvider();
         AuthCache authCache = new BasicAuthCache();
 
         for (AuthCredential authCredential : request.getAuthCredentials()) {
-            UsernamePasswordCredentials userPassword = new UsernamePasswordCredentials(authCredential.getUsername(), authCredential.getPassword());
-            AuthScope scope = AuthScope.ANY;
+            UsernamePasswordCredentials userPassword = new UsernamePasswordCredentials(authCredential.getUsername(), authCredential.getPassword().toCharArray());
+            AuthScope authScope = new AuthScope(null, null, -1, null, null);
             if (authCredential.getHostname() != null) {
-                HttpHost target = HttpHost.create(authCredential.getHostname());
-                scope = new AuthScope(target);
+                HttpHost target;
+                try {
+                    target = HttpHost.create(authCredential.getHostname());
+                    authScope = new AuthScope(target);
+
+                } catch (URISyntaxException e) {
```

Comment:
Is this error handling consistent with what the client expected from V4? For example, a malformed URI what did it do before?

Reply:
v4  and v5 throw a IllegalArgumentException if the host contained blanks.
v5 will now also throw a URISyntaxException if the port is invalid or the scheme contains blanks